src/ecmult_const.h

```diff
@@ -18,4 +18,25 @@
  */
 static void secp256k1_ecmult_const(secp256k1_gej *r, const secp256k1_ge *a, const secp256k1_scalar *q, int bits);
 
+/**
+ * Same as secp256k1_ecmult_const, but takes in an x coordinate of the base point
+ * only, specified as fraction n/d (numerator/denominator). Only the x coordinate of the result is
+ * returned.
+ *
+ * If known_on_curve is 0, a verification is performed that n/d is a valid X
+ * coordinate, and 0 is returned if not. Otherwise, 1 is returned.
+ *
+ * d being NULL is interpreted as d=1.
+ *
+ * Constant time in the value of q, but not any other inputs.
+ */
+static int secp256k1_ecmult_const_xonly(
+    secp256k1_fe* r,
+    const secp256k1_fe *n,
+    const secp256k1_fe *d,
+    const secp256k1_scalar *q,
+    int bits,
+    int known_on_curve
+);
```

Comment:
I think we should mention the `*q == 0` edge case. Right now `secp256k1_ecmult_const_xonly` outputs `*r == 0`, which is maybe a reasonable value in that case but not an X coordinate on the curve. We can either disallow `*q == 0` and VERIFY_CHECK it or document that it outputs invalid X-coordinate 0 (and also test this edge case in the regular tests). I'd prefer the former because outputting an X-coordinate that's not on the curve seems confusing and it relies on our definition that `0^-1 = 0` (see `secp256k1_fe_inv(&i, &i);` in the function).